### ISSUE TITLE
new protocol?  Neje  DK-8-KZ

### ISSUE THREAD

**Issue by firstel** [camrein/EzGraver#53]
I just got the engraver a month or so ago.   I can connect to it, but only the up/down/left/right buttons work.  None of the other bottoms work, and I can not do any engraving..

Could there new a new protocol again?  How can we find out/fix?

**Comment by sewardj1011** [comment]
I am having a similar problem to EllyS215. Would be happy to help provide any information needed to update this so that it works with the latest releases of this unit

**Comment by anabolyc** [comment]
I've got sources from factory v4 tool, and i will work on v4 protocol implementaion
https://github.com/anabolyc/neje-engraver-tool

**Comment by firstel** [comment]
This is great!  So this will be a mac version?  I'm so excited!

**Comment by janoi152** [comment]
I agree, this is great! I hope that the new version will be released soon. Because I just bought NEJE DK-8-KZ engraver recently and I can't use it. The official application is randomly cropping every picture I try to print (which makes it unusable) and EzGraver isn't working since the protocol changed (when I click Start the engraver won't do anything, but the moving and preview functions work).

**Comment by anabolyc** [comment]
I've checked in develop branch to windows version. There i fixed crop bug, also removed call that uploads every image you carve to chinese  servers (yep!), alro removed call that downloads full library from chinese servers before start, so it starts quick now. I'm still planning to implement this in EzGraver but i might take some time

**Comment by janoi152** [comment]
Wow, thank you! That sounds really great and I would like to try it right now but I don't know where are the binaries (executables). I checked your develop branch but there aren't any releases and I can't find any executables. To be honest I was always confused by GitHub (especially when trying to find compiled binaries) and I don't know how to compile that code. What should I do, please? Thank you.

**Comment by janoi152** [comment]
Okay, so I've tried to compile (build) the code in Visual Studio but I'm getting 1 error and 5 warnings. The error is "Form_main does not contain a definition for pictureBox1_Click and no extension method pictureBox1_Click accepting a first argument of type Form_Main could be found (are you missing a using directive or an assembly reference?)". So a compiled version would be really appreciated.

**Comment by anabolyc** [comment]
Sorry, my bad, just sync latest, it is fixed now. And thumbs up for challanging build yourself:)

**Comment by janoi152** [comment]
It works! Thank you very much!

**Comment by firstel** [comment]
Wpw!  Where can I download the updated mac app?

**Comment by tseho** [comment]
hey @anabolyc, do you need help for porting the v4 (and v4.2 ?) protocol to EzGraver ?

**Comment by anabolyc** [comment]
Hi @tseho, unfortunately i'm currently quite busy with other project. I [checked in](https://github.com/anabolyc/EzGraver/commit/b23f4f47ca9fd384c2684b562155d29abede57c9) WIP state to my fork. Engraving part is still not working correctly unfortunately, i have windows version working in parallel and seems that data exchange model changed since v3. You may have a look If you have possibility to run windows version as well.

**Comment by tseho** [comment]
@anabolyc thank you for the work in your fork, I will see if I can make it work.
I will post here if there is any progress.

**Comment by tseho** [comment]
Hello all, after trying to work on @anabolyc code, I gave up after failing to understand where the issue was.
Anyway, I started from scratch and it took me a while but here it is, the "HowTo" send an image to the graver.
https://github.com/tseho/gozgraver/blob/863c0d2feadcd7ff4fc27be0fbcab0cab3dbd32e/core/protocolv4.go#L40
So, if anybody want to reproduce the protocol v4 in EzGraver, I commented as much as possible to explain what's expected.

Currently, [tseho/gozgraver](https://github.com/tseho/gozgraver) is an undocumented command line utility, only tested on linux.
